astropy/modeling/functional_models.py

```diff
@@ -572,6 +572,98 @@ def fit_deriv(x, amplitude, x_0, fwhm):
         return [d_amplitude, d_x_0, d_fwhm]
 
 
+class Voigt1D(Fittable1DModel):
+    """
+    One dimensional model for the Voigt profile.
+
+    Parameters
+    ----------
+    x_0 : float
+        Position of the peak
+    amplitude_L : float
+        The Lorentzian amplitude
+    fwhm_L : float
+        The Lorentzian full width at half maximum
+    fwhm_G : float
+        The Gaussian full width at half maximum
+
+    See Also
+    --------
+    Gaussian1D, Lorentz1D
+
+    Notes
+    -----
+    Algorithm for the computation taken from
+    McLean, A. B., Mitchell, C. E. J. & Swanston, D. M. Implementation of an
+    efficient analytical approximation to the Voigt function for photoemission
+    lineshape analysis. Journal of Electron Spectroscopy and Related Phenomena
+    69, 125-132 (1994)
+
+    Examples
+    --------
+    .. plot::
+        :include-source:
+
+        import numpy as np
+        from astropy.modeling.models import Voigt1D
+        import matplotlib.pyplot as plt
+
+        plt.figure()
+        x = np.arange(0, 10, 0.1)
+        v1 = Voigt1D(x_0=5,amplitude_L=10,fwhm_L=0.5, fwhm_G=0.9)
```

Comment:
One last, minor, truly obnoxious nitpick :)  Could you please add a space after each of the commas between the arguments (after x_0 and amplitude_L that is; fwhm_L already has it).  Since this is example code we want to be especially picky about promoting PEP8-isms.

Other than that this looks great.  For the purposes of this PR don't worry about the fit_deriv issue.  There's nothing wrong with your implementation, and I can't seem to reproduce the problem.  Which is not to say it isn't a problem that I'd still like to get to the bottom of (incidentally, what version of SciPy are you using?)  But whatever the issue is I don't think it's specific to this model.